src/www/app/outline_server_repository/index.ts

```diff
@@ -53,11 +53,20 @@ function isDynamicAccessKey(accessKey: string): boolean {
 // NOTE: For extracting a name that the user has explicitly set, only.
 // (Currenly done by setting the hash on the URI)
 function serverNameFromAccessKey(accessKey: string): string {
-  if (isDynamicAccessKey(accessKey)) {
-    return new URL(accessKey.replace(/^ssconf:\/\//, 'https://')).hostname;
+  const {hostname, hash} = new URL(accessKey.replace(/^ss(?:conf)?:\/\//, 'https://'));
+
+  let result;
+  if (hash && hash !== '#') {
+    // locates the first key-value pair in the search params-compliant
+    // string that has no value (i.e. thing=a&My%20Server%20Name&thing=b)
+    const {groups} = hash.slice(1).match(/(?:^|&)(?<name>[^=]+)(?:&|$)/);
```

Comment:
Can you use URLSearchParams instead? It's easier to read and easier to figure out it's right.

Or split the string first on `&`, then on `=`

Reply:
Favor readability in this case. We are not gaining much from going over the string once.
Why not call `fragment.split("&")` and iterate over it? It will be a lot more readable.